web-ui/src/components/checkin/CheckinHistory.jsx

```diff
@@ -1,120 +1,120 @@
-        setCheckins(checkinList);
-      }
+  const getCheckinDate = () => {
+    if (currentCheckin.checkInDate) {
+      return new Date(currentCheckin.checkInDate);
     }
-    updateCheckins();
-  }, [id]);
+    return new Date();
+  };
 
-  let checkinDate =
-    checkins.length > 0
-      ? new Date(checkins[checkinIndex].checkInDate)
-      : new Date();
   const lastIndex = checkins.length - 1;
-  const leftArrowClass =
-    "arrow " + (checkinIndex < lastIndex ? "enabled" : "disabled");
+  const leftArrowClass = "arrow " + (index > 0 ? "enabled" : "disabled");
   const rightArrowClass =
-    "arrow " + (checkinIndex > 0 ? "enabled" : "disabled");
+    "arrow " + (index < lastIndex ? "enabled" : "disabled");
 
   const previousCheckin = () => {
-    setCheckinIndex((index) => (index === lastIndex ? lastIndex : index + 1));
-    setIndex(checkinIndex);
-    // TODO: change checkin on click
+    if (index !== 0) {
+      setIndex((index) => index - 1);
+    }
   };
 
   const nextCheckin = () => {
-    setCheckinIndex((index) => (index === 0 ? 0 : index - 1));
-    setIndex(checkinIndex);
+    if (index !== lastIndex) {
+      setIndex((index) => index + 1);
+    }
   };
 
-  const pickDate = (date) => {};
+  const pickDate = async (date) => {
+    const year = date.getFullYear();
+    const month = date.getMonth() + 1;
+    const day = date.getDate();
+    const checkin = checkins[index];
+    const dateArray = [year, month, day];
+    const updatedCheckin = await updateCheckin({
+      ...checkin,
+      checkInDate: dateArray,
+    });
+    const newCheckin = updatedCheckin.payload.data;
+    const filtered = checkins.filter((e) => {
+      return e.id !== checkin.id;
+    });
+    filtered.push(newCheckin);
+    dispatch({
+      type: UPDATE_CHECKINS,
+      payload: filtered,
+    });
+    dispatch({
+      type: UPDATE_CURRENT_CHECKIN,
+      payload: newCheckin,
+    });
+  };
 
-  const DateInput = ({ value, onClick }) => (
-    <div className="date-input">
-      <p style={{ margin: "0px" }}>{value}</p>
-      <CalendarTodayIcon onClick={onClick}>stuff</CalendarTodayIcon>
+  const DateInput = React.forwardRef((props, ref) => (
+    <div className="date-input" ref={ref}>
+      <p style={{ margin: "0px" }}>{props.value}</p>
+      <CalendarTodayIcon onClick={props.onClick}>stuff</CalendarTodayIcon>
     </div>
-  );
+  ));
 
   return (
     <div>
-      <div className="profile-section">
-        <Avatar
-          src={imageUrl ? imageUrl : "/default_profile.jpg"}
-          style={{ height: "220px", width: "200px" }}
-        />
-        <div className="info">
-          <p>{name}</p>
-          <p>{role}</p>
-          <p>PDL: {pdl}</p>
-          <p>Company Email: {workEmail}</p>
+      {getCheckinDate() && (
+        <div className="date-picker">
+          <ArrowBackIcon
+            className={leftArrowClass}
+            onClick={previousCheckin}
```

Comment:
My gut on these would be to simply make them links to the appropriate `/checkins/{id}` path rather than using onClick handlers for them...it reduces the number of possible paths for data loading if you are only worried about the current state originating from the URL. That's just my $0.02.